src/optimizer/query_rewrite.c

```diff
@@ -7174,6 +7174,73 @@ qo_move_on_clause_of_explicit_join_to_where_clause (PARSER_CONTEXT * parser, PT_
     }
 }
 
+/*
+ * qo_push_limit_to_union () - push limit on union query
+ *   return: PT_NODE *
+ *   parser(in): parser environment
+ *   node(in): possible query
+ *   arg(in):
+ *   continue_walk(in):
+ */
+static PT_NODE *
+qo_push_limit_to_union (PARSER_CONTEXT * parser, PT_NODE * node, PT_NODE * limit)
+{
+  switch (node->node_type)
+    {
+    case PT_SELECT:
+      if (!pt_has_inst_in_where_and_select_list (parser, node))
+	{
+	  node->info.query.flag.rewrite_limit = 1;
+	  node->info.query.limit = limit;
```

Comment:
Isn't it necessary to guarantee that node->info.query.limit is null when incorporated here?
`assert(node->info.query.limit == NULL);`

Reply:
It is checked in pt_has_inst_in_where_and_select_list (). limit is changed to instnum on semantic check.